### DISCUSSION TITLE
Error: Function Runtimes must have a valid version, for example `now-php@1.0.0`./ The package `vercel-python` is not published on the npm registry

### DISCUSSION THREAD

**Discussion by Vinaykumarvuppala** [vercel/community#7390] [Help] [answered]
### Summary

I am facing below error while deploying:
**Error: Function Runtimes must have a valid version, for example `now-php@1.0.0`.**

used in 18.x Node.js Version
in package.json :  "engines": {
    "node": "18.x"

vercel.json: 
  "functions": {
    "api/app.py": {
      "runtime": "@vercel/python",
      "memory": 1024
    }

tried this method as well:

  "functions": {
    "api/app.py": {
      "runtime": "python@3.9",
      "memory": 1024
    }

### Steps to Reproduce

vercel.json

{
  "version": 2,
  "functions": {
    "api/app.py": {
      "runtime": "@vercel/python",
      "memory": 1024
    }
  },
  "routes": [
    {
      "src": "/(.*)",
      "dest": "api/app.py"
    }
  ]
}

**-----------**

**Comment by pawlean** [comment]
Hi @Vinaykumarvuppala!

Could you try this configuration?

```
{
  "version": 2,
  "functions": {
    "api/app.py": {
      "runtime": "vercel-python@3.9", // or vercel-python@3.12
      "memory": 1024
    }
  },
  "routes": [
    {
      "src": "/(.*)",
      "dest": "api/app.py"
    }
  ]
}

```

**Comment by Vinaykumarvuppala** [comment]
Tried both ways, did not work:
Here is the error message:
Error: Function Runtimes must have a valid version, for example ***@***.***`.

Thanks,

**Comment by satpai1215** [comment]
@pawlean  any update here? Does Python deployment currently work for Vercel? I am getting the same error with the following configuration (my project is currently on Node 20.x)

**Error: Function Runtimes must have a valid version, for example `now-php@1.0.0`.**

```
{
    "version": 2,
    "functions": {
      "api/index.py": {
        "runtime": "python@3.12"
      }
    },
    "routes": [
      {
        "src": "/api/(.*)",
        "dest": "api/index.py"
      }
    ]
  }
  ```

**Comment by TooTallNate** [comment, marked as answer]
You don't need the "runtime" property at all if you are trying to deploy Python - it's one of the built-in languages for Vercel. In fact, unless you are trying to override some of the other configuration options (like `memory` or `maxDuration`), then there is no need to have that "functions" object at all. You can delete it.

[a comment by an automated account is not shown]
